test/persistence-hooks.suite.js

```diff
@@ -2556,6 +2556,45 @@ module.exports = function(dataSource, should, connectorCapabilities) {
             });
         });
 
+        it('applies updates from `persist` hook on create', function(done) {
+          TestModel.observe('persist', ctxRecorder.recordAndNext(function(ctx) {
+            ctx.data.extra = 'hook data';
+          }));
+
+          // By default, the instance passed to updateAttributes callback is NOT updated
+          // with the changes made through persist/loaded hooks. To preserve
+          // backwards compatibility, we introduced a new setting updateOnLoad,
+          // which if set, will apply these changes to the model instance too.
+          TestModel.settings.forceId = true;
+          TestModel.settings.updateOnLoad = true;
+          TestModel.replaceOrCreate(
+            {name: 'a name'}, function(err, instance) {
+            if (err) return done(err);
+            instance.should.have.property('extra', 'hook data');
+            TestModel.settings.updateOnLoad = false;
```

Comment:
This code will not be executed when the assertion above fails.

Fortunately, there is no need to reset this flag, because a fresh new `TestModel` class is created for each test (see `beforeEach` at the top of this test file).

Same comment applies to the other new test added below.